### PR TITLE
[spec] Change 'schedule a postTask task' to actually return the promise

### LINKED ISSUE #42: schedule a postTask task doesn't always return a promise
In particular step 8 doesn't seem to return anything.

### PR DESCRIPTION
**Pull request by shaseley**
This closes #42 (thanks @annevk!).

@domenic is it acceptable to word it like this, or should this be a separate step?

### CODE REVIEW

File: spec/scheduling-tasks.md [hunk 1]
```diff
@@ -265,7 +265,8 @@ Processing Model {#sec-scheduling-tasks-processing-model}
     1. [=Schedule a task to invoke a callback=] for |scheduler| given |queue|,
        |signal|, |callback|, and |result|.
   1. Otherwise the task is not delayed. [=Schedule a task to invoke a callback=]
-     for |scheduler| given |queue|, |signal|, |callback|, and |result|.
+     for |scheduler| given |queue|, |signal|, |callback|, and |result|, then
```
Comment (annevk) [inline spec/scheduling-tasks.md:268]:
```suggestion
     for |scheduler| given |queue|, |signal|, |callback|, and |result|, and then
```

**Review by annevk — APPROVED** [review]
This is reasonable. The alternative which is slightly cleaner I think would be something like:

1. If ..., run these steps in parallel:
   1. ...
1. Otherwise, schedule a task to ...
1. Return _result_.

**Comment by domenic** [comment]
+1 to this being good and also to @annevk's suggestion being nicer :)

**Comment by shaseley** [comment]
Let's do the nicer thing :). Thanks all!

**Comment by annevk** [comment]
You also want to remove the return statement in step 7 as it's redundant.

**Comment by shaseley** [comment]
> You also want to remove the return statement in step 7 as it's redundant.

Ahh, yep, sorry for missing that! Yeah this is cleaner with the one return path, thanks!

**Review by domenic — APPROVED** [review]

**Pull request merged**